app/code/Magento/CatalogImportExport/Model/Import/Product.php

```diff
@@ -1756,6 +1823,11 @@ protected function _saveProducts()
                     }
                 }
 
+                // 5.1 Items to remove phase
+                if (isset($existingImages[$rowSku])) {
+                    $galleryItemsToRemove = \array_diff(\array_keys($existingImages[$rowSku]), $uploadedFiles);
```

Comment:
Your code only removes the images of the last product because the ```$galleryItemsToRemove``` variable is reassigned for each product.

I propose the use of ```array_merge``` as follows:
```
$galleryItemsToRemove = \array_merge($galleryItemsToRemove, \array_diff(\array_keys($existingImages[$rowSku]), $uploadedFiles));
```